ConfigSpace/configuration_space.py

```diff
@@ -48,34 +50,35 @@ class ConfigurationSpace(object):
     """Represent a configuration space.
     """
 
-    def __init__(self, seed=None):
-        self._hyperparameters = OrderedDict()
-        self._hyperparameter_idx = dict()
-        self._idx_to_hyperparameter = dict()
+    def __init__(self, seed: Union[int, None] = None) -> None:
+        self._hyperparameters = OrderedDict()  # type: OrderedDict[str, Hyperparameter]
+        self._hyperparameter_idx = dict()  # type: Dict[str, int]
+        self._idx_to_hyperparameter = dict()  # type: Dict[int, str]
 
         # Use dictionaries to make sure that we don't accidently add
         # additional keys to these mappings (which happened with defaultdict()).
         # This once broke auto-sklearn's equal comparison of configuration
         # spaces when _children of one instance contained  all possible
         # hyperparameters as keys and empty dictionaries as values while the
         # other instance not containing these.
-        self._children = OrderedDict()
-        self._parents = OrderedDict()
+        self._children = OrderedDict()   # type: OrderedDict[str, OrderedDict[str, Union[None, AbstractCondition]]]
+        self._parents = OrderedDict()   # type: OrderedDict[str, OrderedDict[str, Union[None, AbstractCondition]]]
 
         # changing this to a normal dict will break sampling because there is
         #  no guarantee that the parent of a condition was evaluated before
-        self._conditionsals = OrderedDict()
-        self.forbidden_clauses = []
+        self._conditionsals = OrderedDict()   # type: OrderedDict[str, str]
```

Comment:
line 169 is 
**self._conditionsals[child_node] = child_node**
from here it appears that key and value are equal. debugger also says the same